contents/tutorials/track-high-volume-apis.md

```diff
@@ -0,0 +1,266 @@
+        },
+      },
+    );
+  }
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+You might notice that this sends a "Feature Flag Called" event to PostHog on every call. We can turn this off by adding `{ sendFeatureFlagEvents: false }` to our `client.isFeatureEnabled()` call.
+
+```node
+//...
+const sample = await client.isFeatureEnabled(
+	'sample',
+	req.params.id,
+	{ sendFeatureFlagEvents: false }
+);
+//...
+```
+
+### Locally evaluating our flag
+
+Another problem is that every feature flag called on the server side sends a request to PostHog to evaluate the flag. Over a high volume of API calls, this small difference can add up. We can limit this by locally evaluating the flag with person properties the flag relies on.
+
+To do this, add `firstRouteCalled` in `personProperties` to the options in our `client.isFeatureEnabled()` call.
+
+```node
+//...
+const sample = await client.isFeatureEnabled(
+    'sample',
+    req.params.id,
+    { 
+      sendFeatureFlagEvents: false,
+      personProperties: {
+        'firstRouteCalled': req.url,
+      },
+    }
+  );
+//...
+```
+
+In my small example, this cut response time by about 10ms, and every millisecond counts for high-volume APIs.
+
+## 2. Cache usage and batch events
+
+Another way to capture events for high-volume APIs is to cache usage of the API and then capture a batch of events at a time interval. 
+
+To do this, we use [node-cache](https://www.npmjs.com/package/node-cache) to create a basic cache to store the number of requests, then use [node-cron](https://www.npmjs.com/package/node-cron) to set up a cron job to read the cache, capture values for each of the IDs, and clear the cache. 
+
+First, install them both.
+
+```bash
+npm i node-cache node-cron
+```
+
+Next, set up our cache which simply gets the current value for the param ID and then increments it.
+
+```node
+// index.js
+//...
+const NodeCache = require('node-cache');
+const cache = new NodeCache()
+
+//...
+
+app.get('/big/:id', async (req, res) => {
+  let calls = cache.get(req.params.id) || 0;
+  calls++;
+  cache.set(req.params.id, calls);
+
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+After, we can set up our cron job to loop through the keys in the cache, get the number of events for that key, capture a batch event with a distinct ID (we use later), then clear the cache once completed.
+
+```node
+// index.js
+//...
+const cron = require('node-cron');
+
+//...
+
+cron.schedule('*/10 * * * * *', () => {
+  cache.keys().forEach((key) => {
+    let calls = cache.get(key);
+    client.capture(
+      { 
+        event: 'big route batch',
+        distinctId: key,
+        properties: { 
+          calls: calls,
+					distinctId: key
+        },
+      },
+    );
+  });
+  cache.flushAll();
+});
+```
+
+Once you run this and send multiple events to different IDs, you see batch events with calls as a property.
```

Comment:
This is unclear. What do you mean by "send multiple events to different IDs" ?
```suggestion
Once you run this and send multiple events to different IDs, you see batch events with the API calls as a property.
```

Reply:
Reworded